File: content/en/04-llms/llms/_exercises/12_plot-image-1/12_plot-image-1.R
library(readr)
library(ellmer)
library(ggplot2)

# Step 1: Make a scatter plot of the typical mtcars dataset
mtcars <- read_csv(here::here("data/mtcars.csv"))

ggplot(mtcars) +
  aes(x = wt, y = mpg) +
  geom_point(color = "steelblue", size = 3) +
  labs(
    title = "MPG vs Weight",
    x = "Weight (1000 lb)",
    y = "Miles per Gallon (mpg)"
  ) +
  theme_bw()

# Step 2: Ask the model to interpret the plot.
# (Hint: see `content_image_...`)
chat <- chat_posit(model = "____", echo = "output")
chat$chat(
  "Interpret this plot.",
  ____()
)

File: content/en/04-llms/llms/_exercises/12_plot-image-1/12_plot-image-1.py
# %%
import chatlas
import polars as pl
from matplotlib import pyplot as plt
from plotnine import aes, geom_point, ggplot, labs, theme_bw
from pyhere import here

# %% [markdown]
# Step 1: Create a scatter plot of `mpg` vs `wt` from the `mtcars` dataset using
# `plotnine`.

# %%
mtcars = pl.read_csv(here("data/mtcars.csv"))
mtcars

# %%
p = (
    ggplot(mtcars, aes(x="wt", y="mpg"))
    + geom_point(color="steelblue", size=2)
    + labs(title="MPG vs Weight", x="Weight (1000 lb)", y="Miles per Gallon (mpg)")
    + theme_bw()
)

p.show()

# %%
# Register the plot with matplotlib's current figure
plt.figure(p.draw())

# %% [markdown]
# Step 2: Ask the model to interpret the plot.
# (Hint: see `content_image_...`)

# %%
chat = chatlas.ChatPosit(model="____")
chat.chat(
    "Interpret this plot.",
    chatlas.____(),
)
